pyk/src/pyk/integration_tests/kprove_test.py

```diff
@@ -28,7 +28,9 @@ def setUp(self):
 
         self.kprove = KProve(self.kompiled_dir, kprove_main_file, Path(self.KPROVE_USE_DIR))
         self.kprove.prover_args += list(chain.from_iterable(['-I', include_dir] for include_dir in kprove_include_dirs))
-        self._update_symbol_table(self.kprove.symbol_table)
+        # force computation of the symbol_table before updating it
+        if self.kprove.symbol_table:
+            self._update_symbol_table(self.kprove._symbol_table)
```

Comment:
To me it looks like the original version achieves the same thing. Did you run into issues with that?

Reply:
Yeah, calling `self.kprove.symbol_table` forces it to actually load the definition, so that `_symbol_table` is not `None` anymore. Then we can patch it.

Reply:
But doesn't `self._update_symbol_table(self.kprove.symbol_table)` also evaluate the property before passing the value to the function? (Sorry, I might be missing some detail.)

Reply:
Hmmm, that's a good point, but for some reason it wasn't working. I'll play with it a bit more and see if I can get it to work.

Reply:
Ah no, the original code doesn't work actually. Because it passes `self.kprove.symbol_table`, which is now a `@property` instead of a member field, and then `_patch_symbol_table` tries to update that property which is not allowed.

Instead here, we call the property to make sure it's oladed the first time, then we're patching on the `_symbol_table` underlying field.

Reply:
That's strange. I tried it now to see what's going on, but it didn't break integration tests. Maybe there's some discrepancy between Python 3.8 and 3.10.

Reply:
I tried the original version again, and it worked! I am not sure what happened.

Though, it does seem strange to me that it did work honestly. Why would be be able to write to a `@property` directly?

Reply:
Good question. This is a read-only property, which indeed means it only has a getter, but no setter. But in this case, the setter is not invoked - what is called is `kprove.symbol_table['key'] = value`, which conceptually translates to `kprove.get_symbol_table().__setitem__('key', value)`, so it's fine.

Reply:
Nice, makes sense. Thank you.